features/FxA-15-OAuth-Management/README.md

```diff
@@ -0,0 +1,89 @@
+Disconnect OAuth services from FxA
+======================
+
+https://github.com/mozilla/fxa-features/issues/20
+
+## Problem Summary
+
+As a Firefox Accounts user I want to be able to revoke access to the
+services that I logged into.  
+
+****
+
+## Outcomes
+
+The success criteria of this feature:
+* Being able to view logged in OAuth services in the apps view.
+* Being able to revoke access ("disconnect") a service.
+
+See "Detailed Design" below for more details.
+
+## Hypothesis
+
+Being able to disconnect the services from user settings
+will improve the quality and security of our service.
+This functionality is a core feature for services that provide OAuth relier login.
+We will know this is true when we see activity in the "Connected apps and devices"
+
+## Metrics
+
+We would collect usage metrics with StatsD:
+* number of views of the apps and devices view,
+* number of revoke actions for any service.
+* number of revoke actions for each service by client id.
+
+Derive a percentage of settings views vs apps view to understand
+the popularity of this feature.
+
+****
+
+## Detailed design
+
+<img src='pr_moz_mock.png' width='440' />
+
+The mock up above adds "apps" functionality to the
+[Devices View (FXA-89)](https://github.com/mozilla/fxa/pull/181/files).
+
+The initial OAuth apps list would
+revoke the non-expired tokens using the web UI.
+It will be displayed to logged in users of FxA
+in the "Devices & apps" settings view.
+This new functionality changes the current devices view and adds to it.
+
+The list will also include the sessions that were created
+for the Firefox Accounts website. Firefox Browser and Sync logins
+are treated as 'device clients' and are managed as 'Devices', not apps.
+
+* The dashboard will order the services by last login to a given service.
+* The OAuth servers calls the listed items "clients". To make it easier to understand
+we call them "apps" in the settings view.
+* The button to remove all active sessions will say "Disconnect". To match "devices".
+* The dashboard will provide a "last active" date in the same format as the devices view.
+* The dashboard's title in settings will be "Devices & apps".
+* All services will have the same default service icon to improve the visual UX of the feature.
+* Revoking Access:
+  * OAuth service: will revoke all non-expired tokens belonging to a given OAuth service.
+  * Only the OAuth services that do not need permission to get implicit grants will be listed (`canGrant=0`).
```

Comment:
>  Is it because we expect them to show up as active sessionTokens in this view?

Yeap exactly. Added this after chatting with @seanmonstar about it.